airflow/api_internal/endpoints/rpc_api_endpoint.py

```diff
@@ -68,7 +70,10 @@ def internal_airflow_api(
 
     log.debug("Calling method %.", {method_name})
     try:
-        output = handler(**params)
+        output = handler(
+            **params,
+            log=logging.getLogger(f"airflow.internal_api.{method_name}"),
```

Comment:
Does it work if the method doesn't have "log" parameter?

Reply:
Oh! That's a good point! I guess not. I dont think we should handle logs this way then. The simplest way would be to create a logger `log = logging.getLogger(__name__)` in `airflow/dag_processing/manager.py` and use this logger in static methods. The only downside is the logger would be the same whether or not internal API in turn on

Reply:
I hoped that we can detect that the method has "log" argument and add it then.

If it's not possible to detect then  we can send addition parameter from client(like "inject_log=true") and when seen on server-side then add "log" to params

Reply:
I implemented a solution which detects if the parameter "log" exists. If it does, then provide one specific to internal API

Reply:
This looks great! Thanks!